File: 24/python/twentyfour.py
def factorial(n):
	if n == 1:
		return 1
	return n*factorial(n-1)

def solution(p, n, digits):
	if n == 0:
		return digits 
	thisdigit = digits[p/factorial(n)]
	digits.pop(p/factorial(n))
	return [thisdigit]+solution(p%factorial(n), n-1, digits)

digits = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
print solution(1000000, 9, digits)	

File: 24/java/Twentyfour.java
import java.util.*;
public class Twentyfour
{
	public static void main(String[] args)
	{
		LinkedList<Integer> digits = new LinkedList<Integer>();
		for(int i=0; i<10; i++)
		{
			digits.add(i);
		}
		System.out.println(solution(1000000, 9, digits).toString());
	}
	public static int factorial(int n)
	{
		if (n == 1)
			return 1;
		return n*factorial(n-1);
	}
	public static LinkedList<Integer> solution (int p, int n, LinkedList<Integer> digits)
	{
		LinkedList<Integer> result = new LinkedList<Integer>();
		while (n > 0)
		{
			result.add(digits.get(p/factorial(n)));
			digits.remove(p/factorial(n));
			p=p%factorial(n);
			n--;
		}
		result.add(digits.get(digits.size()-1));
		return result;
	}
}